opencog/atoms/core/Variables.cc

```diff
@@ -1272,36 +1272,34 @@ void Variables::extend(const Variables& vset)
 {
 	for (const Handle& h : vset.varseq)
 	{
-		try
+		auto index_it = index.find(h);
+		if (index_it != index.end())
 		{
-			index.at(h);
-
 			// Merge the two typemaps, if needed.
-			try
+			auto typemap_it = vset._simple_typemap.find(h);
+			if (typemap_it != vset._simple_typemap.end())
 			{
-				const TypeSet& tms = vset._simple_typemap.at(h);
+				const TypeSet& tms = typemap_it->second;
 				TypeSet mytypes =
 					type_intersection(_simple_typemap[h], tms);
 				_simple_typemap.erase(h);	 // is it safe to erase if
                                              // h not in already?
 				_simple_typemap.insert({h, mytypes});
 			}
-			catch(const std::out_of_range&) {}
 		}
-		catch(const std::out_of_range&)
```

Comment:
well, but these changes make the code run slower.  In general, `try..catch` is always faster than `if(error) {}`

The point is that you always pay the code of `if(error)` even if there is no error -- and the cose is not just the if test, but a possible icache miss and/or itlb miss, which can be expensive. By contrast, `try..catch` always costs zero extra cycles/cache-misses, unless there is an error. Its faster, smalllerr, cheaper, easier and more fool-proof -- this is why exceptions were added to C++...